subsys/bluetooth/host/classic/ssp.c

```diff
@@ -213,6 +213,14 @@ static int ssp_confirm_neg_reply(struct bt_conn *conn)
 
 static void ssp_pairing_complete(struct bt_conn *conn, uint8_t status)
 {
+	/* When the ssp pairing complete event notified,
+	 * clear the pairing flag.
+	 */
+	atomic_clear_bit(conn->flags, BT_CONN_BR_PAIRING);
+	atomic_set_bit(conn->flags, BT_CONN_BR_PAIRED);
```

Comment:
If `status != 0` you should probably not set the `PAIRED` flag, right?

Reply:
Right. But here `BT_CONN_BR_PAIRED` is just used to flag the BR pairing is done.
If the pairing is failed, the link encrypted event should be notified. And then `cross transport procedure` should not be started.
But I think maybe there is a corner case that the connection has been paired before. And in this connection, user want to increase security level. But the pairing is failed. And then the  connection can also be encrypted by using previous link key. In this case, the flag `BT_CONN_BR_PAIRED` has been set. Then `cross transport procedure` will be started after link encrypted event notified. But `cross transport procedure` should not be started in this case.

I will improve it.

Reply:
Updated.